File: 05-valid-palindrome/valid-palindrome.py
# https://leetcode.com/problems/valid-palindrome/
from re import sub
class Solution:
    def isPalindrome(self, s: str) -> bool:
        palindrome = sub(r'[^A-Za-z0-9]*', '', s).lower()
        i = 0
        j = len(palindrome) - 1
        while (i < j):
            if (palindrome[i] != palindrome[j]):
                return False
            i += 1
            j -= 1
        return True


print(Solution().isPalindrome('socorram me subi no onibus em marrocos'))
# print(Solution().isPalindrome("A man, a plan, a canal: Panama"))

File: 05-valid-palindrome/valid-palindrome.js
// https://leetcode.com/problems/valid-palindrome/
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
    const palindrome = s.replace(/[^a-zA-Z0-9]*/g, '').toLocaleLowerCase();
    let i = 0;
    let j = palindrome.length - 1;
    while (i < j) {
        if (palindrome[i] !== palindrome[j]) {
            return false;
        }
        i++;
        j--;
    }
    return true;
};

// regex to remove everything
// everything to localelowercase
// get first and last letter and check if they're equal

console.log(isPalindrome('socorram me subi no onibus em marrocos'));
console.log(isPalindrome('arara'));
console.log(isPalindrome('arra'));
